pkg/ruler/remotequerier.go

```diff
@@ -69,6 +72,16 @@ func (c *QueryFrontendConfig) RegisterFlags(f *flag.FlagSet) {
 			"to enable client side load balancing.")
 
 	c.GRPCClientConfig.RegisterFlagsWithPrefix("ruler.query-frontend.grpc-client-config", f)
+
+	f.StringVar(&c.QueryResultResponseFormat, "ruler.query-frontend.query-result-response-format", formatJSON, fmt.Sprintf("Format to use when retrieving query results from query-frontends. Supported values: %s", strings.Join(allFormats, ", ")))
```

Comment:
I'm not 100% sure whether this flag should go under `-ruler.query-frontend....` or directly under `-ruler....` - could be convinced either way, open to opinions.